packages/typography/src/utility/StaticWidthText.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import Box from '@leafygreen-ui/box';
+import { getNodeTextContent, HTMLElementProps } from '@leafygreen-ui/lib';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+type StaticWidthTextProps<T extends keyof JSX.IntrinsicElements> =
+  HTMLElementProps<T> & {
+    /**
+     * The maximum future weight of the text. Determines the width of the component.
+     * @default 700
+     */
+    maxFontWeight?: React.CSSProperties['fontWeight'];
+    /**
+     * Defines the pseudo element used to force the element width
+     */
+    pseudoElement?: 'before' | 'after';
+    as?: T;
+  };
+
+const staticWidthTextStyle = ({
+  pseudoElement,
+  maxFontWeight,
+}: Pick<StaticWidthTextProps<any>, 'pseudoElement' | 'maxFontWeight'>) => {
+  const pseudoSelector = `&:${pseudoElement}`;
+
+  return css`
+    /* Inherit relevant properties from the parent */
+    font-weight: inherit;
+    font-size: inherit;
```

Comment:
Not related to the component, but do you prefer splitting up your CSS based on properties?

Reply:
Yeah, I try to follow (roughly) the categories in [idiomatic-css](https://github.com/necolas/idiomatic-css#declaration-order)